bayes_opt/bayesian_optimization.py

```diff
@@ -165,6 +167,10 @@ def maximize(self,
         util = UtilityFunction(kind=acq, kappa=kappa, xi=xi)
         iteration = 0
         while not self._queue.empty or iteration < n_iter:
+            # decay kappa by decay factor if BO iters exceeds kappa_decay_delay
+            if kappa_decay != 1 and max(0, len(self.res) - init_points) - kappa_decay_delay > 0:
```

Comment:
I rather not introduce this `if` statement here. It's at a different abstraction level than the surrounding code and it also only applies to one kind of utility function.

Reply:
is this still something you wanted to move forward with?

Reply:
Yes, absolutely. I've been a bit busy lately with work, so I'm slow to make comments and stuff. But we'll get this merged this week =)